### PR TITLE
2D unclosed sketch Mirror

### PR DESCRIPTION
**Pull request by jessfraz**
this takes some of the feed back from https://github.com/KittyCAD/modeling-app/issues/3959 since we needed it anyways

### CODE REVIEW

File: src/wasm-lib/kcl/src/std/revolve.rs [hunk 1]
```diff
@@ -25,65 +25,57 @@ pub struct RevolveData {
...
-            RevolveAxisAndOrigin::Z => ([0.0, 0.0, 1.0], [0.0, 0.0, 0.0]),
-            RevolveAxisAndOrigin::NegX => ([-1.0, 0.0, 0.0], [0.0, 0.0, 0.0]),
-            RevolveAxisAndOrigin::NegY => ([0.0, -1.0, 0.0], [0.0, 0.0, 0.0]),
-            RevolveAxisAndOrigin::NegZ => ([0.0, 0.0, -1.0], [0.0, 0.0, 0.0]),
-            RevolveAxisAndOrigin::Custom { axis, origin } => (*axis, *origin),
+            AxisAndOrigin::X => ([1.0, 0.0, 0.0], [0.0, 0.0, 0.0]),
+            AxisAndOrigin::Y => ([0.0, 1.0, 0.0], [0.0, 0.0, 0.0]),
+            AxisAndOrigin::NegX => ([-1.0, 0.0, 0.0], [0.0, 0.0, 0.0]),
+            AxisAndOrigin::NegY => ([0.0, -1.0, 0.0], [0.0, 0.0, 0.0]),
+            AxisAndOrigin::Custom { axis, origin } => ([axis[0], axis[1], 0.0], [origin[0], origin[1], 0.0]),
```
Comment (nadr0) [inline src/wasm-lib/kcl/src/std/revolve.rs:78]:
```
 Custom {
        /// The axis.
        axis: [f64; 2],
        /// The origin.
        origin: [f64; 2],
    },
```

Why do these have 3 coordinates? Should these just be 2?

Comment (jessfraz) [reply src/wasm-lib/kcl/src/std/revolve.rs:78]: engine expects three, we are merely fixing user errors

Comment (nadr0) [reply src/wasm-lib/kcl/src/std/revolve.rs:78]: nvm this is correct.

**Review by nadr0 — APPROVED** [review]
yarp, also thanks for tacking on that axis issue!

**Pull request merged**